websites/Z/ZARP Gaming/presence.ts

```diff
@@ -0,0 +1,141 @@
+const presence = new Presence({
+	clientId: "990710812320612432",
+});
+
+presence.on("UpdateData", () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "zarp-512x512",
+	};
+	switch (window.location.host) {
+		case "zarpgaming.com":
+			if (
+				window.location.pathname == "/index.php" ||
+				window.location.pathname == "/index.php/forum"
+			)
+				presenceData.details = "Viewing the home page";
+
+			if (window.location.pathname.startsWith("/index.php/forum/")) {
+				presenceData.details = "Viewing a section:";
+				presenceData.state = (document.querySelector(
+					"#Kunena > form > div > div.kheader > h2 > span"
+				) as HTMLElement)?.innerText || document.title;
```

Comment:
don't use `as` and don't use HTMLElement, use the actual class name
```suggestion
				presenceData.state = document.querySelector<HTMLElement>(
					"#Kunena > form > div > div.kheader > h2 > span"
				)?.innerText ?? document.title;
```